src/ReportManager/DetailsSection/SchemaForm/fields/Text/index.js

```diff
@@ -26,13 +26,13 @@ const TEXT_INPUT_TYPE_TO_INPUT = { [INPUT_TYPE.SHORT]: ShortTextInput, [INPUT_TY
 const TEXT_INPUT_TYPE_STYLES = { [INPUT_TYPE.SHORT]: styles.shortText, [INPUT_TYPE.LONG]: styles.longText };
 
 const Text = ({ id }) => {
-  const { fields, formData, onFieldChange } = useContext(SchemaFormContext);
+  const { fields, fieldErrors, fieldValues, onFieldChange } = useContext(SchemaFormContext);
```

Comment:
Now we read from the context three flat objects: `fields`, `fieldErrors` and `fieldValues`. They map the UI details, validation error and current value of each field by their ids. This makes accessing this information trivial and O(1) since we don't have to handle recursivity with collections.